WebExample/tests/textManipulation.spec.ts

```diff
@@ -0,0 +1,148 @@
+  });
+
+  test('paste redo', async ({page}) => {
+    const PASTE_TEXT_FIRST = '*bold*';
+    const PASTE_TEXT_SECOND = '@here';
+
+    const inputLocator = await setupInput(page, 'clear');
+
+    await page.evaluate(async (pasteText) => navigator.clipboard.writeText(pasteText), PASTE_TEXT_FIRST);
+    await inputLocator.press(`${OPERATION_MODIFIER}+v`);
+    await page.waitForTimeout(CONSTANTS.INPUT_HISTORY_DEBOUNCE_TIME_MS);
+    await page.evaluate(async (pasteText) => navigator.clipboard.writeText(pasteText), PASTE_TEXT_SECOND);
+    await page.waitForTimeout(CONSTANTS.INPUT_HISTORY_DEBOUNCE_TIME_MS);
+    await inputLocator.press(`${OPERATION_MODIFIER}+v`);
+    await page.waitForTimeout(CONSTANTS.INPUT_HISTORY_DEBOUNCE_TIME_MS);
+
+    await inputLocator.press(`${OPERATION_MODIFIER}+z`);
+    await inputLocator.press(`${OPERATION_MODIFIER}+Shift+z`);
+
+    expect(await inputLocator.innerText()).toBe(`${PASTE_TEXT_FIRST}${PASTE_TEXT_SECOND}`);
+  });
+});
+
+test('select', async ({page}) => {
+  const inputLocator = await setupInput(page, 'reset');
+  await inputLocator.focus();
+  await inputLocator.press(`${OPERATION_MODIFIER}+a`);
+
+  const cursorPosition = await page.evaluate(() => {
+    const editableDiv = document.querySelector('div[contenteditable="true"]');
+    const range = window.getSelection()?.getRangeAt(0);
+    if (!range || !editableDiv) return null;
+    const preCaretRange = range.cloneRange();
+    preCaretRange.selectNodeContents(editableDiv);
+    preCaretRange.setEnd(range.endContainer, range.endOffset);
+
+    return preCaretRange.toString().length;
+  });
+
+  expect(cursorPosition).toBe(CONSTANTS.EXAMPLE_CONTENT.length);
+});
+
+test('cut content changes', async ({page}) => {
+  const INITIAL_CONTENT = 'bold';
+  const WRAPPED_CONTENT = CONSTANTS.MARKDOWN_STYLE_DEFINITIONS.bold.wrapContent(INITIAL_CONTENT);
+  const EXPECTED_CONTENT = CONSTANTS.MARKDOWN_STYLE_DEFINITIONS.bold.wrapContent(INITIAL_CONTENT).slice(0, 3);
+
+  const inputLocator = await setupInput(page, 'clear');
+  await pasteContent({text: WRAPPED_CONTENT, page, inputLocator});
+  const rootHandle = await inputLocator.locator('span.root').first();
+
+  await page.evaluate(async (initialContent) => {
+    const filteredNode = Array.from(document.querySelectorAll('div[contenteditable="true"] > span.root span')).find((node) => {
+      return node.textContent?.includes(initialContent) && node.nextElementSibling && node.nextElementSibling.textContent?.includes('*');
+    });
+
+    const startNode = filteredNode;
+    const endNode = filteredNode?.nextElementSibling;
+
+    if (startNode?.firstChild && endNode?.lastChild) {
+      const range = new Range();
+      range.setStart(startNode.firstChild, 2);
+      range.setEnd(endNode.lastChild, endNode.lastChild.textContent?.length ?? 0);
+
+      const selection = window.getSelection();
+      selection?.removeAllRanges();
+      selection?.addRange(range);
+    }
```

Comment:
Setting cursor position can also be moved to utils

Reply:
we probably won't be able to use it in here, because this would be only a part of the function used inside of page.evaluate and we aren't able to pass functions to the interior, as a param. I'll add this to the utils anyway, in case you're ever in need of using it somewhere else.